asapdiscovery-docking/asapdiscovery/docking/openeye.py

```diff
@@ -0,0 +1,375 @@
+"""
+This module contains the inputs, docker, and output schema for using POSIT
+"""
+import abc
+from enum import Enum
+from pathlib import Path
+from typing import Literal, Optional, Union
+
+import dask
+from asapdiscovery.data.dask_utils import actualise_dask_delayed_iterable
+from asapdiscovery.data.openeye import (
+    combine_protein_ligand,
+    oechem,
+    oedocking,
+    oeomega,
+    save_openeye_pdb,
+)
+from asapdiscovery.data.schema_v2.ligand import Ligand, compound_names_unique
+from asapdiscovery.data.schema_v2.pairs import DockingInputPair
+from asapdiscovery.docking.docking_data_validation import (
+    DockingResultColsV2 as DockingResultCols,
+)
+from asapdiscovery.docking.docking_v2 import (
+    DockingBase,
+    DockingInputsBase,
+    DockingResult,
+)
+from asapdiscovery.modeling.modeling import split_openeye_design_unit
+from pydantic import BaseModel, Field, PositiveFloat, PositiveInt, root_validator
+
+
+class POSIT_METHOD(Enum):
+    """
+    Enum for POSIT methods
+    """
+
+    ALL = oedocking.OEPositMethod_ALL
+    HYBRID = oedocking.OEPositMethod_HYBRID
+    FRED = oedocking.OEPositMethod_FRED
+    MCS = oedocking.OEPositMethod_MCS
+    SHAPEFIT = oedocking.OEPositMethod_SHAPEFIT
+
+    @classmethod
+    def reverse_lookup(cls, value):
+        return cls(value).name
+
+
+class POSIT_RELAX_MODE(Enum):
+    """
+    Enum for POSIT relax modes
+    """
+
+    CLASH = oedocking.OEPoseRelaxMode_CLASHED
+    ALL = oedocking.OEPoseRelaxMode_ALL
+    NONE = oedocking.OEPoseRelaxMode_NONE
+
+
+class POSITInputs(DockingInputsBase):
```

Comment:
We have moved away from using input Schema as separate classes/objects. It is a bit too decoupled, and makes more sense for openeye because of C++ stuff/  Instead these can just be attributes of the workflow class itself. You can share via inheritance as you have done here also.

Reply:
I will say the exception to this is as an argument to a function that needs a tonne of validation.

Reply:
I am confused because you have a LargeScaleDockingInputs class (which is where I modeled this from):
https://github.com/choderalab/asapdiscovery/blob/3839851460c93f0555f3044fdfd0c92f27b94a12/asapdiscovery-docking/asapdiscovery/docking/workflows/large_scale_docking.py#L37-L250

I don't have strong opinions either way but I think we should be consistent with implementation?

Reply:
I think if we have results schema it makes some sense to have inputs schema

Reply:
Ah okay perhaps the naming and API design is a bit confusing, lets discuss tommorow. But basically:

The LargeScaleDockingInputs thing is to validate a workflow, ie collect a tonne of information, and validation to run a **function** that does the whole workflow, stopping a situation like this. 

```python
workflow_func(a=1, b=2, c=3, d=42, e=... # takes a lot of information that then needs to be reconciled. )
```
Instead with 

```python
workflow_func(input_schema) 
```
The idea is that the function doesn't have to do any sanitation of the inputs. 

For **classes** that run steps (components), I don't see the need to separate the settings and object itself, as the object can easily contain its own settings if that makes sense? All the current component classes use the second pattern below roughly, but perhaps I need to formalize this in a document.

The only difference between 

```python
class DockingInputs 
    a=1
    b=2
    # use a, b validation in class directly
```
 and 

```python
class Docker
    inputs = DockingInputs()
   ...
   # use inputs.a, inputs.b validation in DockingInputs
```

Is that there are two classes instead of one and its more nested. Both could work though to be perfectly clear